peps/pep-0746.rst

```diff
@@ -0,0 +1,166 @@
+PEP: 746
+Title: TypedMetadata for type checking of PEP 593 Annotated
+Author: Adrian Garcia Badaracco <adrian@adriangb.com>
+Sponsor: Jelle Zijlstra <jelle.zijlstra@gmail.com>
+Discussions-To: https://discuss.python.org/t/XYZ
+Status: Draft
+Type: Standards Track
+Topic: Typing
+Created: 20-May-2024
+Python-Version: 3.13
+Post-History: 20-May-2024
+
+Abstract
+--------
+
+This PEP proposes a mechanism for integrating metadata in the :pep:`593` ``Annotated`` type
+into the type system.
+
+Motivation
+----------
+
+:pep:`593` introduced ``Annotated`` as a way to attach runtime metadata to types,
+however it does not provide any way to tie the metadata to the type itself.
+This can introduce type unsoundness in real world usage, even in examples similar
+to those in the PEP.
+For example, given a metadata like ``Int64()`` that might instruct how to serialize a field
+over the wire, one can write ``Annotated[float, Int64()]`` which is clearly unsound
+and could be type checked, but there is currently no mechanism to do so.
+
+Runtime type checking libraries
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+One of the core desires for :pep:`593` was to allow better integration with runtime
+uses of types. It proposed a use case similar to the above ``Int64`` example. However,
+it has become clear that one of the most popular use cases for typing in Python is runtime
+type checking, as provided by libraries like `pydantic <pydantic_>`_ and `msgspec <msgspec>`_.
```

Comment:
We can fix the "Duplicate explicit target name" errors and simplify these links (because the name and target are the same) like this:

```suggestion
type checking, as provided by libraries like `pydantic`_ and `msgspec`_.
```